Uchu.Core/Serializable/Struct/Property/RequiredPacketProperty.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections.Generic;
+using System.Linq;
+using RakDotNet.IO;
+
+namespace Uchu.Core
+{
+    public class RequiredPacketProperty : WrappedPacketProperty
+    {
+        /// <summary>
+        /// Values for the property that are required to write.
+        /// </summary>
+        public readonly List<object> RequiredValues = new List<object>();
+
+        /// <summary>
+        /// Property that is checked.
+        /// </summary>
+        private readonly string _propertyToCheck;
+        
+        /// <summary>
+        /// Creates the required packet property.
+        /// </summary>
+        /// <param name="packetProperty">Packet property to wrap.</param>
+        /// <param name="propertyToCheck">Property that is checked.</param>
+        public RequiredPacketProperty(IPacketProperty packetProperty, string propertyToCheck) : base(packetProperty)
+        {
+            this._propertyToCheck = propertyToCheck;
+        }
+        
+        /// <summary>
+        /// Writes the property to the writer.
+        /// </summary>
+        /// <param name="objectToWrite">Object with the property to write.</param>
+        /// <param name="writer">Bit writer to write to.</param>
+        /// <param name="writtenProperties">Properties that were previously written.</param>
+        public override void Write(object objectToWrite, BitWriter writer, Dictionary<string, object> writtenProperties)
+        {
+            if (!writtenProperties.ContainsKey(this._propertyToCheck))
```

Comment:
Here and on line 53 could be a check that ensures `writtenProperties` / `readProperties` is not null (as you have done for NiQuaternion and String properties)

Reply:
Simple enough change and should remove 2 compile warnings.